### PR TITLE
Dat 3641

### PR DESCRIPTION
**Pull request by rybmat**

### CODE REVIEW

File: extensions/wikia/Search/templates/WikiaSearch_CrossWiki_result.php [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <?php
 	// get wiki thumbnail and thumbnail tracking
 	$isOnWikiMatch = isset($result['onWikiMatch']) && $result['onWikiMatch'];
-	$imageFileName = PromoImage::fromPathname($result['image_s'])->ensureCityIdIsSet($result['id'])->getPathname();
+	$commData = new CommunityDataService($result['id']);
```
Comment (idradm) [inline extensions/wikia/Search/templates/WikiaSearch_CrossWiki_result.php]: As we are changing this, maybe we could move it out of template to controller?

**Comment by idradm** [comment]
What about description? Do we want to leave fallback to old approach in case of new data missing, if not what should be used instead?

File: extensions/wikia/Search/templates/WikiaSearch_CrossWiki_index.php [hunk 2]
```diff
@@ -30,14 +30,8 @@
...
-						echo $app->getView('WikiaSearch', 'CrossWiki_result', array(
-							'result' => $result,
-							'pos' => $pos + (($currentPage - 1) * $resultsPerPage),
-							'query' => $query,
-							'hub' => $hub,
-							'corporateWikiId' => $corporateWikiId,
-							'wgExtensionsPath' => $wgExtensionsPath
-						));
+						echo $app->getView('WikiaSearch', 'CrossWiki_result',
+							\Wikia\Search\Result\ResultHelper::extendResult($result, $pos + (($currentPage - 1) * $resultsPerPage), 60));
```
Comment (sebastianmarzjan) [inline extensions/wikia/Search/templates/WikiaSearch_CrossWiki_index.php]: Let's make 16 and 60 consts

File: extensions/wikia/Search/classes/Test/Result/ResultHelperTest.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+namespace Wikia\Search\Test\ResultSet;
+use \WikiaBaseTest as WikiaBaseTest, Wikia\Search\Result\ResultHelper;
+
+class AbstractResultSetTest extends WikiaBaseTest {
+	public function testLimitTextLength() {
```
Comment (sebastianmarzjan) [inline extensions/wikia/Search/classes/Test/Result/ResultHelperTest.php]: Let us add a few more cases. In the end, it makes sense to have this trimming in a separate class (not related to search results in any way)

File: extensions/wikia/Search/classes/Test/Result/ResultHelperTest.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+namespace Wikia\Search\Test\ResultSet;
+use \WikiaBaseTest as WikiaBaseTest, Wikia\Search\Result\ResultHelper;
+
+class AbstractResultSetTest extends WikiaBaseTest {
```
Comment (sebastianmarzjan) [inline extensions/wikia/Search/classes/Test/Result/ResultHelperTest.php]: The name needs to change to reflect what this test does AND prevent fatals during test execution

File: extensions/wikia/Search/classes/Result/ResultHelper.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			'videoMsg' => $service->shortnumForMsg( $result['videos_i']?:0, 'wikiasearch2-videos' ),
+			'title' => ( $sn = $result->getText( 'sitename_txt' ) ) ? $sn : $result->getText( 'headline_txt' ),
+			'url' => $result->getText( 'url' ),
+			'hub' => $result->getHub(),
+			'pos' => $pos
+		];
+	}
+
+	public static function limitTextLength($text, $wordLimit = null) {
+		$textAsString = is_array( $text ) ? implode( " ", $text ) : $text;
```
Comment (sebastianmarzjan) [inline extensions/wikia/Search/classes/Result/ResultHelper.php]: Why is this line needed?
